src/services/sendEmailService.ts

```diff
@@ -46,8 +46,59 @@ var sendEmailToAgents = async (shifts, emailToAgent) => {
+}
+
+
+function getReportingMessage(attachment,email) {
+  let content = `Hi ${email}! Please find your report in the attachment.`
+
+  return {
+    to: email,
+    from: {
+      email: 'supportinternal@mail.zoovu.io',
+      name: 'Support App'
+    },
+    subject: 'Your purchase events report',
+    text: content,
+    attachments: [
+      {
+        content: attachment,
+        filename: "purchaseReport.csv",
+        type: "text/csv",
+        disposition: "attachment"
+      }
+    ]
+  };
+}
+
+async function sendReportingEmail(attachment,email) {
+  const message = getReportingMessage(attachment,email);
+  try {
+    await sendGridMail.send(message);
+    console.log(`Reporting email sent successfully to ${message.to}`);
+  } catch (error) {
+    console.error(`Error sending reporting email to ${message.to}`);
+    if (error.response) {
+      console.error(error.response.body.errors)
+    }
+  }
+}
+
+var sendReporting = async (rowsCSV, email) => {
+  //return if email for a given agent was not provided
+  if (!email) return;
+
+  sendGridMail.setApiKey(process.env.SENDGRIDAPIKEY);
+
+  const fs = require('fs');
+  let attachment = Buffer.from(rowsCSV).toString('base64');
+  await sendReportingEmail(attachment,email);
+}
+
+export default sendEmailToAgents;
+export {
+  sendReporting
 }
```

Comment:
please move all of that above generateHTML function and use/adjust it to send the reporting email as well.
if we already have an HTML template in line with the support messages from zendesk then lets use it